### DISCUSSION TITLE
Accidental push of a large file now I'm stuck

### DISCUSSION THREAD

**Discussion by georgem1956** [community/community#63388] [New to GitHub]
I accidentally committed a large file in my local branch and pushed it to my github repository. Of course it failed due to its size. How do I remove it from my local branch? Even though I've deleted the file my push thinks it's still there and tries to upload it to my repository.

**Comment by georgem1956** [comment]
I did a git reset soft to a to a commit prior to adding this file; showed the file in git status; removed the file; commit those changes and it's still wants to be pushed to my repository.

**Comment by anztim** [comment]
`git reset` might help you to reverse a commit operation. You can learn details about how to use it on the internet.

If the commit did not successfully push to GitHub repository, there is an uncomplicated way to go back to the situation which before you commit that file.

1. Delete all local files including the directory.
2. Clone the repository from GitHub with `git clone`

**Comment by DavidSmitMri** [comment]
This worked for me: 

- Locally delete large files.
- Commit the local deletes.
- Soft reset back X number of commits (two commits before I pushed the large file): git reset --soft HEAD~2.
- Then recommit all the changes together (AKA squash) git commit -m "New message for the combined commit"
- Push squashed commit.
git push --force

https://stackoverflow.com/a/49703619/1184603

**Reply by deckerc23** [reply]
Thank you! I have been stuck so long and this worked for me yay

**Reply by prwkhar** [reply]
Thank you

**Reply by CodeMstr14Trihaji** [reply]
this is works! 🤯 thanks man

**Reply by schecters5150** [reply]
My man thank you so fucking much - you just save me a lot of work here. Crisis averted at 12am

**Reply by ifeoluwaayo** [reply]
works. thanks

**Comment by jerryjames2001** [comment]
Hi!  I understand how frustrating that can be — Git can be tricky with large files.

Even if you’ve deleted the file and committed the deletion, Git history still holds that large file in earlier commits. That’s why your push keeps failing — Git is trying to upload the file from your commit history.

Here’s how you can completely remove the file from Git history:

# Steps to Remove the File from History:

1. Run this command to remove the file from all commits:
```
git filter-branch --force --index-filter "git rm --cached --ignore-unmatch path/to/largefile" --prune-empty --tag-name-filter cat -- --all
```
Replace path/to/largefile with the actual file path.

2. Then clean up the reflog and garbage:
```
rm -rf .git/refs/original/
git reflog expire --expire=now --all
git gc --prune=now
```
3. Force push the cleaned history:
```
git push origin --force --all
```
⚠️ Warning: This rewrites Git history, so make sure you're not working in a shared branch (like main) or coordinate with your team before doing this.

# If you don’t mind starting fresh:

You could also delete your local repo folder and git clone a clean copy again (as user2 mentioned), but that won’t remove the large file if it’s already in remote history.
